lib/facter/puppet_config.rb

```diff
@@ -0,0 +1,30 @@
+desired_settings = {
+  master: %i[
+    localcacert
+    ssldir
+  ],
+  main: %i[
+    hostpubkey
+    hostprivkey
+    hostcert
+    localcacert
+    ssldir
+    vardir
+    server
+  ]
+}
+Facter.add(:puppet_config) do
+  confine kernel: 'Linux'
```

Comment:
Why confine it for Linux only? It seems generic for other platforms as well?

Reply:
Thanks, i have created another #151  to remove this